src/components/BlockDocumentInput.vue

```diff
@@ -5,23 +5,31 @@
     <template v-if="blockDocuments.length">
       <BlockDocumentCombobox v-model:selected="model" v-bind="{ blockDocuments }" class="block-document-input__select" />
     </template>
-    <p-button v-if="blockTypeSlug" icon-append="PlusIcon" :to="withRedirect(routes.blockCreate(blockTypeSlug))">
+    <p-button v-if="blockTypeSlug && useModal" icon-append="PlusIcon" @click="open">
       Add
     </p-button>
+    <p-button v-else-if="blockTypeSlug" icon-append="PlusIcon" :to="withRedirect(routes.blockCreate(props.blockTypeSlug))">
+      Add
+    </p-button>
```

Comment:
Won't blockTypeSlug always be defined? Its a required prop 🤔 (I know this if was here before. But seems like it might have been left over from a previous iteration).

Reply:
Hmmm... good point!